com.hazelcast.jet.sql.impl.connector.jdbc.mysql.MySQLAllTypesInsertJdbcSqlConnectorTest.insertRowWithAllTypes [HZ-2298]

_master_ (commit 765cc352c833d7f27d3a438c18a4027b0a3158aa)

Failed on oracle-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OracleJDK8-nightly/943/testReport/com.hazelcast.jet.sql.impl.connector.jdbc.mysql/MySQLAllTypesInsertJdbcSqlConnectorTest/

Failed in variants:
- `[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]`
- `[type:REAL, mappingType:REAL, value:1.5, expected:1.5]`

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 

Expecting actual:
  [Row{[0, 1.5]}, Row{[1, 1.5]}]
to contain exactly in any order:
  [Row{[0, 1.5]}, Row{[1, 1.5]}]
elements not found:
  [Row{[0, 1.5]}, Row{[1, 1.5]}]
and elements not expected:
  [Row{[0, 1.5]}, Row{[1, 1.5]}]

	at com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport.assertJdbcRowsAnyOrder(JdbcSqlTestSupport.java:194)
	at com.hazelcast.jet.sql.impl.connector.jdbc.AllTypesInsertJdbcSqlConnectorTest.insertRowWithAllTypes(AllTypesInsertJdbcSqlConnectorTest.java:109)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)

```
</details>


<details><summary>Standard output:</summary>

```
Finished Running Test: insertRowWithAllTypes[type:DECIMAL (10,5), mappingType:DECIMAL, sqlValue:1.12345, javaValue:1.123, jdbcValue:1.123] in 0.054 seconds.
Started Running Test: insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]
08:20:01,509 DEBUG |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [CalciteSqlOptimizer] insertRowWithAllTypes - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Before logical opt:
LogicalTableInsert(table=[[hazelcast, public, mapping_o_1aafabc4_b152_42d1_8dcc_b1beee5a6342[projects=[$0, $1]]]], operation=[INSERT], flattened=[false])
  LogicalValues(tuples=[[{ 0, 1.5E0 }]])

08:20:01,509 DEBUG |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [CalciteSqlOptimizer] insertRowWithAllTypes - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] After logical opt:
InsertLogicalRel(table=[[hazelcast, public, mapping_o_1aafabc4_b152_42d1_8dcc_b1beee5a6342[projects=[$0, $1]]]], operation=[INSERT], flattened=[false])
  ValuesLogicalRel(values=[{expressions=[[ConstantExpression{type=QueryDataType {family=INTEGER}, value=0}, ConstantExpression{type=QueryDataType {family=REAL}, value=1.5}]]}])

08:20:01,510 DEBUG |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [CalciteSqlOptimizer] insertRowWithAllTypes - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] After physical opt:
InsertPhysicalRel(table=[[hazelcast, public, mapping_o_1aafabc4_b152_42d1_8dcc_b1beee5a6342[projects=[$0, $1]]]], operation=[INSERT], flattened=[false])
  ValuesPhysicalRel(values=[{expressions=[[ConstantExpression{type=QueryDataType {family=INTEGER}, value=0}, ConstantExpression{type=QueryDataType {family=REAL}, value=1.5}]]}])

08:20:01,513 DEBUG |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [LightMasterContext] insertRowWithAllTypes - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Start executing light job 09b1-6212-d60e-0001, execution graph in DOT format:
digraph DAG {
	"Values" [localParallelism=1];
	"Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])" [localParallelism=2];
	"Values" -> "Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])" [label="isolated", queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com to visualize the printed graph.
08:20:01,513 DEBUG |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [LightMasterContext] insertRowWithAllTypes - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Building execution plan for 09b1-6212-d60e-0001
08:20:01,513 DEBUG || - [LightMasterContext] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Built execution plans for 09b1-6212-d60e-0001
08:20:01,513 DEBUG || - [InitExecutionOperation] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Initializing execution plan for job 09b1-6212-d60e-0001, execution 09b1-6212-d60e-0001 from [127.0.0.1]:5701
08:20:01,514 DEBUG || - [JobExecutionService] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Execution plan for light job ID=09b1-6212-d60e-0001, jobName='09b1-6212-d60e-0001', executionId=09b1-6212-d60e-0001 initialized, will start the execution
08:20:01,514 DEBUG || - [InitExecutionOperation] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_romantic_wu.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Initializing execution plan for job 09b1-6212-d60e-0001, execution 09b1-6212-d60e-0001 from [127.0.0.1]:5701
08:20:01,514 DEBUG || - [WriteJdbcP] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.jet.blocking.thread-0 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] [09b1-6212-d60e-0001/Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])#1] Actual pool size used: 1
08:20:01,514 DEBUG || - [WriteJdbcP] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.jet.blocking.thread-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] [09b1-6212-d60e-0001/Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])#0] Actual pool size used: 1
08:20:01,515 DEBUG || - [JobExecutionService] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_romantic_wu.cached.thread-2 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Execution plan for light job ID=09b1-6212-d60e-0001, jobName='09b1-6212-d60e-0001', executionId=09b1-6212-d60e-0001 initialized, will start the execution
08:20:01,515 DEBUG || - [WriteJdbcP] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_romantic_wu.jet.blocking.thread-1 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] [09b1-6212-d60e-0001/Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])#2] Actual pool size used: 1
08:20:01,515 DEBUG || - [WriteJdbcP] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_romantic_wu.jet.blocking.thread-2 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] [09b1-6212-d60e-0001/Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])#3] Actual pool size used: 1
08:20:01,515 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Completed execution of job '09b1-6212-d60e-0001', execution 09b1-6212-d60e-0001
08:20:01,515 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Execution of job '09b1-6212-d60e-0001', execution 09b1-6212-d60e-0001 completed
08:20:01,520 DEBUG || - [WriteJdbcP] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_romantic_wu.jet.blocking.thread-1 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] [09b1-6212-d60e-0001/Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])#2] release null
08:20:01,520 DEBUG || - [JobExecutionService] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_romantic_wu.cached.thread-2 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Completed execution of job '09b1-6212-d60e-0001', execution 09b1-6212-d60e-0001
08:20:01,520 DEBUG || - [JobExecutionService] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_romantic_wu.cached.thread-2 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Execution of job '09b1-6212-d60e-0001', execution 09b1-6212-d60e-0001 completed
08:20:01,525 DEBUG |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [CalciteSqlOptimizer] insertRowWithAllTypes - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Before logical opt:
LogicalTableInsert(table=[[hazelcast, public, mapping_o_1aafabc4_b152_42d1_8dcc_b1beee5a6342[projects=[$0, $1]]]], operation=[INSERT], flattened=[false])
  LogicalCalc(expr#0=[{inputs}], expr#1=[1], expr#2=[?0], id=[$t1], table_column=[$t2])
    LogicalValues(tuples=[[{ 0 }]])

08:20:01,526 DEBUG |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [CalciteSqlOptimizer] insertRowWithAllTypes - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] After logical opt:
InsertLogicalRel(table=[[hazelcast, public, mapping_o_1aafabc4_b152_42d1_8dcc_b1beee5a6342[projects=[$0, $1]]]], operation=[INSERT], flattened=[false])
  ValuesLogicalRel(values=[{projection=[ConstantExpression{type=QueryDataType {family=INTEGER}, value=1}, ParameterExpression{index=0}], values=[{expressions=[[ConstantExpression{type=QueryDataType {family=INTEGER}, value=0}]]}]}])

08:20:01,527 DEBUG |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [CalciteSqlOptimizer] insertRowWithAllTypes - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] After physical opt:
InsertPhysicalRel(table=[[hazelcast, public, mapping_o_1aafabc4_b152_42d1_8dcc_b1beee5a6342[projects=[$0, $1]]]], operation=[INSERT], flattened=[false])
  ValuesPhysicalRel(values=[{projection=[ConstantExpression{type=QueryDataType {family=INTEGER}, value=1}, ParameterExpression{index=0}], values=[{expressions=[[ConstantExpression{type=QueryDataType {family=INTEGER}, value=0}]]}]}])

08:20:01,528 DEBUG |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [LightMasterContext] insertRowWithAllTypes - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Start executing light job 09b1-6212-d60f-0001, execution graph in DOT format:
digraph DAG {
	"Values" [localParallelism=1];
	"Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])" [localParallelism=2];
	"Values" -> "Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])" [label="isolated", queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com to visualize the printed graph.
08:20:01,528 DEBUG |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [LightMasterContext] insertRowWithAllTypes - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Building execution plan for 09b1-6212-d60f-0001
08:20:01,529 DEBUG || - [LightMasterContext] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Built execution plans for 09b1-6212-d60f-0001
08:20:01,529 DEBUG || - [InitExecutionOperation] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Initializing execution plan for job 09b1-6212-d60f-0001, execution 09b1-6212-d60f-0001 from [127.0.0.1]:5701
08:20:01,530 DEBUG || - [JobExecutionService] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Execution plan for light job ID=09b1-6212-d60f-0001, jobName='09b1-6212-d60f-0001', executionId=09b1-6212-d60f-0001 initialized, will start the execution
08:20:01,530 DEBUG || - [InitExecutionOperation] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_romantic_wu.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Initializing execution plan for job 09b1-6212-d60f-0001, execution 09b1-6212-d60f-0001 from [127.0.0.1]:5701
08:20:01,531 DEBUG || - [WriteJdbcP] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.jet.blocking.thread-0 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] [09b1-6212-d60f-0001/Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])#0] Actual pool size used: 1
08:20:01,531 DEBUG || - [WriteJdbcP] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.jet.blocking.thread-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] [09b1-6212-d60f-0001/Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])#1] Actual pool size used: 1
08:20:01,531 DEBUG || - [JobExecutionService] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_romantic_wu.cached.thread-1 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Execution plan for light job ID=09b1-6212-d60f-0001, jobName='09b1-6212-d60f-0001', executionId=09b1-6212-d60f-0001 initialized, will start the execution
08:20:01,531 DEBUG || - [WriteJdbcP] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_romantic_wu.jet.blocking.thread-1 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] [09b1-6212-d60f-0001/Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])#2] Actual pool size used: 1
08:20:01,531 DEBUG || - [WriteJdbcP] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_romantic_wu.jet.blocking.thread-2 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] [09b1-6212-d60f-0001/Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])#3] Actual pool size used: 1
08:20:01,532 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-1 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Completed execution of job '09b1-6212-d60f-0001', execution 09b1-6212-d60f-0001
08:20:01,532 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-1 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Execution of job '09b1-6212-d60f-0001', execution 09b1-6212-d60f-0001 completed
08:20:01,533 DEBUG || - [WriteJdbcP] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.jet.blocking.thread-0 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] [09b1-6212-d60f-0001/Insert([table_o_6c9da65b_f2d7_4727_8a58_303a8b8b039a])#0] release null
08:20:01,533 DEBUG || - [JobExecutionService] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Completed execution of job '09b1-6212-d60f-0001', execution 09b1-6212-d60f-0001
08:20:01,533 DEBUG || - [JobExecutionService] hz.MySQLAllTypesInsertJdbcSqlConnectorTest_blissful_wu.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Execution of job '09b1-6212-d60f-0001', execution 09b1-6212-d60f-0001 completed
08:20:01,950  INFO |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [SqlTestSupport] Thread-295 - Removing 0 cached plans in SqlTestSupport.@After
08:20:01,950  INFO |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [SqlTestSupport] Thread-295 - Removing 0 cached plans in SqlTestSupport.@After
08:20:01,952  INFO |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [SimpleTestInClusterSupport] Thread-295 - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
08:20:01,952  INFO |insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5]| - [SimpleTestInClusterSupport] Thread-295 - Destroying 1 distributed objects in SimpleTestInClusterSupport.@After: [hz:impl:mapService/__sql.catalog]
BuildInfo right after insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5](com.hazelcast.jet.sql.impl.connector.jdbc.mysql.MySQLAllTypesInsertJdbcSqlConnectorTest): BuildInfo{version='5.3.0-SNAPSHOT', build='20230412', buildNumber=20230412, revision=765cc35, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'insertRowWithAllTypes[type:REAL, mappingType:REAL, sqlValue:1.5, javaValue:1.5, jdbcValue:1.5](com.hazelcast.jet.sql.impl.connector.jdbc.mysql.MySQLAllTypesInsertJdbcSqlConnectorTest):'
08:20:00, accumulated pauses: 411 ms, max pause: 397 ms, pauses over 1000 ms: 0


No metrics recorded during the test

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-OracleJDK8-nightly/943/

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/mysql/MySQLGenericMapStoreTest.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/mysql/MySQLGenericMapStoreTest.java
@@ -18,16 +18,11 @@ package com.hazelcast.mapstore.mysql;
 
 import com.hazelcast.mapstore.GenericMapStoreTest;
 import com.hazelcast.test.annotation.NightlyTest;
-import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.jdbc.MySQLDatabaseProvider;
 import org.junit.BeforeClass;
 import org.junit.experimental.categories.Category;
 
-import java.util.Arrays;
-
-import static java.util.stream.Collectors.joining;
-
-@Category({NightlyTest.class, ParallelJVMTest.class})
+@Category(NightlyTest.class)
 public class MySQLGenericMapStoreTest extends GenericMapStoreTest {
 
     @BeforeClass
@@ -35,10 +30,4 @@ public class MySQLGenericMapStoreTest extends GenericMapStoreTest {
         initialize(new MySQLDatabaseProvider());
     }
 
-    @Override
-    protected String quote(String... parts) {
-        return Arrays.stream(parts)
-                     .map(part -> '`' + part.replaceAll("`", "``") + '`')
-                     .collect(joining("."));
-    }
 }

--- a/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java
+++ b/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java
@@ -25,6 +25,7 @@ import com.hazelcast.jet.core.Vertex;
 import com.hazelcast.jet.impl.util.Util;
 import com.hazelcast.jet.sql.impl.connector.HazelcastRexNode;
 import com.hazelcast.jet.sql.impl.connector.SqlConnector;
+import com.hazelcast.jet.sql.impl.connector.jdbc.mysql.HazelcastMySqlDialect;
 import com.hazelcast.spi.impl.NodeEngine;
 import com.hazelcast.sql.impl.QueryException;
 import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
@@ -37,6 +38,7 @@ import com.hazelcast.sql.impl.type.QueryDataType;
 import org.apache.calcite.rex.RexNode;
 import org.apache.calcite.sql.SqlDialect;
 import org.apache.calcite.sql.SqlDialectFactoryImpl;
+import org.apache.calcite.sql.SqlDialects;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
@@ -49,6 +51,7 @@ import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
 
@@ -249,7 +252,15 @@ public class JdbcSqlConnector implements SqlConnector {
         JdbcDataConnection dataConnection = getAndRetainDataConnection(nodeEngine, dataConnectionName);
         try (Connection connection = dataConnection.getConnection()) {
             DatabaseMetaData databaseMetaData = connection.getMetaData();
-            SqlDialect dialect = SqlDialectFactoryImpl.INSTANCE.create(databaseMetaData);
+            SqlDialect dialect;
+            switch (databaseMetaData.getDatabaseProductName().toUpperCase(Locale.ROOT).trim()) {
+                case "MYSQL":
+                    dialect = new HazelcastMySqlDialect(SqlDialects.createContext(databaseMetaData));
+                    break;
+
+                default:
+                    dialect = SqlDialectFactoryImpl.INSTANCE.create(databaseMetaData);
+            }
             SupportedDatabases.logOnceIfDatabaseNotSupported(databaseMetaData);
             return dialect;
         } catch (Exception e) {

--- a/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/mysql/HazelcastMySqlDialect.java
+++ b/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/mysql/HazelcastMySqlDialect.java
@@ -1,0 +1,62 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.jdbc.mysql;
+
+import com.hazelcast.jet.sql.impl.validate.operators.string.HazelcastConcatOperator;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.SqlWriter.Frame;
+import org.apache.calcite.sql.SqlWriter.FrameTypeEnum;
+import org.apache.calcite.sql.dialect.MysqlSqlDialect;
+
+/**
+ * Custom dialect for MySQL which allows correct unparsing of MySQL specific operators, like CONCAT
+ */
+public class HazelcastMySqlDialect extends MysqlSqlDialect {
+
+    /**
+     * Creates a HazelcastMySqlDialect.
+     */
+    public HazelcastMySqlDialect(Context context) {
+        super(context);
+    }
+
+    @Override
+    public void unparseCall(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+        switch (call.getKind()) {
+            case OTHER:
+                if (call.getOperator() instanceof HazelcastConcatOperator) {
+                    unparseConcat(writer, call);
+                }
+                break;
+
+            default:
+                super.unparseCall(writer, call, leftPrec, rightPrec);
+        }
+    }
+
+    private void unparseConcat(SqlWriter writer, SqlCall call) {
+        writer.print("CONCAT");
+        Frame frame = writer.startList(FrameTypeEnum.PARENTHESES, "(", ")");
+        for (SqlNode operand : call.getOperandList()) {
+            writer.sep(",");
+            operand.unparse(writer, 0, 0);
+        }
+        writer.endList(frame);
+    }
+}

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/DeleteJdbcSqlConnectorTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/DeleteJdbcSqlConnectorTest.java
@@ -160,7 +160,7 @@ public class DeleteJdbcSqlConnectorTest extends JdbcSqlTestSupport {
 
     @Test
     public void deleteFromWithQuotedColumnInWhere() throws Exception {
-        createTable(tableName, "\"person-id\" INT PRIMARY KEY", "name VARCHAR(100)");
+        createTable(tableName, quote("person-id") + " INT PRIMARY KEY", "name VARCHAR(100)");
         insertItems(tableName, 1);
 
         execute(

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlTestSupport.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlTestSupport.java
@@ -40,7 +40,6 @@ import java.util.Properties;
 
 import static com.hazelcast.test.DockerTestUtil.assumeDockerEnabled;
 import static java.util.Objects.requireNonNull;
-import static java.util.stream.Collectors.joining;
 import static org.assertj.core.api.Assertions.assertThat;
 
 /**
@@ -94,9 +93,7 @@ public abstract class JdbcSqlTestSupport extends SqlTestSupport {
     }
 
     protected String quote(String... parts) {
-        return Arrays.stream(parts)
-                     .map(part -> '\"' + part.replaceAll("\"", "\"\"") + '\"')
-                     .collect(joining("."));
+        return databaseProvider.quote(parts);
     }
 
     /**

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/UpdateJdbcSqlConnectorTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/UpdateJdbcSqlConnectorTest.java
@@ -356,7 +356,7 @@ public class UpdateJdbcSqlConnectorTest extends JdbcSqlTestSupport {
 
     @Test
     public void updateMappingWithQuotedColumnInWhere() throws Exception {
-        createTable(tableName, "\"person-id\" INT PRIMARY KEY", "name VARCHAR(100)");
+        createTable(tableName, quote("person-id") + " INT PRIMARY KEY", "name VARCHAR(100)");
         insertItems(tableName, 1);
 
         execute("CREATE MAPPING " + tableName + " DATA CONNECTION " + TEST_DATABASE_REF);
@@ -369,7 +369,7 @@ public class UpdateJdbcSqlConnectorTest extends JdbcSqlTestSupport {
 
     @Test
     public void updateMappingWithQuotedColumnInSet() throws Exception {
-        createTable(tableName, "id INT PRIMARY KEY", "\"full-name\" VARCHAR(100)");
+        createTable(tableName, "id INT PRIMARY KEY", quote("full-name") + " VARCHAR(100)");
         insertItems(tableName, 1);
 
         execute("CREATE MAPPING " + tableName + " DATA CONNECTION " + TEST_DATABASE_REF);

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/mysql/MySQLAllTypesInsertJdbcSqlConnectorTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/mysql/MySQLAllTypesInsertJdbcSqlConnectorTest.java
@@ -38,6 +38,11 @@ public class MySQLAllTypesInsertJdbcSqlConnectorTest extends AllTypesInsertJdbcS
         assumeThat(type).describedAs("TIMESTAMP WITH TIME ZONE not supported on MySQL")
                 .isNotEqualTo("TIMESTAMP WITH TIME ZONE");
 
+        // MySQL REAL type is by default a synonym for DOUBLE PRECISION
+        // MySQL uses FLOAT as 4-byte floating point type
+        if (type.equals("REAL")) {
+            type = "FLOAT";
+        }
     }
 
 }

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/mysql/MySQLAllTypesSelectJdbcSqlConnectorTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/mysql/MySQLAllTypesSelectJdbcSqlConnectorTest.java
@@ -43,5 +43,10 @@ public class MySQLAllTypesSelectJdbcSqlConnectorTest extends AllTypesSelectJdbcS
         assumeThat(type).describedAs("TIMESTAMP WITH TIME ZONE not supported on MySQL")
                 .isNotEqualTo("TIMESTAMP WITH TIME ZONE");
 
+        // MySQL REAL type is by default a synonym for DOUBLE PRECISION
+        // MySQL uses FLOAT as 4-byte floating point type
+        if (type.equals("REAL")) {
+            type = "FLOAT";
+        }
     }
 }

--- a/hazelcast/src/test/java/com/hazelcast/test/jdbc/MySQLDatabaseProvider.java
+++ b/hazelcast/src/test/java/com/hazelcast/test/jdbc/MySQLDatabaseProvider.java
@@ -18,6 +18,10 @@ package com.hazelcast.test.jdbc;
 
 import org.testcontainers.jdbc.ContainerDatabaseDriver;
 
+import java.util.Arrays;
+
+import static java.util.stream.Collectors.joining;
+
 public class MySQLDatabaseProvider implements TestDatabaseProvider {
 
     private static final int LOGIN_TIMEOUT = 120;
@@ -37,5 +41,13 @@ public class MySQLDatabaseProvider implements TestDatabaseProvider {
             ContainerDatabaseDriver.killContainer(jdbcUrl);
             jdbcUrl = null;
         }
+    }
+
+    @Override
+    public String quote(String[] parts) {
+        return Arrays.stream(parts)
+                     .map(part -> '`' + part.replaceAll("`", "``") + '`')
+                     .collect(joining("."));
+
     }
 }

--- a/hazelcast/src/test/java/com/hazelcast/test/jdbc/TestDatabaseProvider.java
+++ b/hazelcast/src/test/java/com/hazelcast/test/jdbc/TestDatabaseProvider.java
@@ -19,8 +19,10 @@ package com.hazelcast.test.jdbc;
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.SQLException;
+import java.util.Arrays;
 
 import static com.hazelcast.internal.util.Preconditions.checkState;
+import static java.util.stream.Collectors.joining;
 
 /**
  * Database provider allows changing database used in a test by providing
@@ -53,4 +55,13 @@ public interface TestDatabaseProvider {
      * Stops the database
      */
     void shutdown();
+
+    /**
+     * Quote individual parts of a compound identifier and concat with `.` delimiter
+     */
+    default String quote(String[] parts) {
+        return Arrays.stream(parts)
+                     .map(part -> '\"' + part.replaceAll("\"", "\"\"") + '\"')
+                     .collect(joining("."));
+    };
 }

